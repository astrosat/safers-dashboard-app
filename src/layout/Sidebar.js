import React from 'react';

import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import logoPng from 'assets/images/background-light-logo@3x.png';
import logo from 'assets/images/fire-white.png';

import SidebarContent from './SidebarContent';

const Sidebar = props => {
  return (
    <React.Fragment>
      <div className="vertical-menu">
        <div className="navbar-brand-box">
          <Link to="/" className="logo logo-light" aria-label="logo">
            <span className="logo-sm">
              <img src={logo} alt="Small logo" height="35" />
            </span>
            <span className="logo-lg">
              <img src={logoPng} alt="Large logo" height="60" />
            </span>
          </Link>
        </div>
        <div data-simplebar className="h-100">
          {props.type !== 'condensed' ? <SidebarContent /> : <SidebarContent />}
        </div>
        <div className="sidebar-background"></div>
      </div>
    </React.Fragment>
  );
};

Sidebar.propTypes = {
  type: PropTypes.string,
};

const mapStatetoProps = state => {
  return {
    layout: state.Layout,
  };
};
export default connect(mapStatetoProps, {})(Sidebar);
