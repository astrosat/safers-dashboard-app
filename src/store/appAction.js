export {
  signIn,
  signUp,
  signOut,
  reqResetPsw,
  resetPsw,
  isRemembered,
  signInOauth2,
  signUpOauth2
} from './authentication/action';
export {
  setDefaultAoi,
  setAoiSuccess,
  getInfo,
  updateInfo,
  uploadProfImg,
  deleteAccount,
  resetProfilePsw
} from './user/action'
export {
  getOrgList,
  getRoleList,
  getAllAreas,
  generalInProgress,
  getConfig
} from './common/action'
export {
  getAllEventAlerts,
  setEventParams,
  setNewEventState,
  resetEventAlertsResponseState,
  getEventInfo,
  editEventAlertInfo,
  setFilteredEventAlerts,
  setEventFavoriteAlert
} from './events/action'
export {
  getSource,
  getAllFireAlerts,
  setFilteredAlerts,
  setFavoriteAlert,
  validateAlert,
  editAlertInfo,
  setAlertApiParams,
  setNewAlertState,
  resetAlertsResponseState
} from './alerts/action'
export {
  getMetaData,
  resetMetaData,
  getAllDataLayers,
  resetDataLayersResponseState,
  postMapRequest,
  getAllMapRequests,
  setNewMapRequestState,
  getDataLayerTimeSeriesData,
} from './datalayer/action'
export {
  getAllInSituAlerts,
  resetInSituAlertsResponseState,
  setCurrentPage,
  setFilteredInSituAlerts,
  setPaginatedAlerts,
  getCameraList,
  getCameraSources,
  getCamera,
  setInSituFavoriteAlert
} from './insitu/action'
export {
  getInSituMedia,
  getStats,
  getTweets,
  getWeatherStats,
  getWeatherVariables
} from './dashboard/action'
export {
  getAllNotifications,
  getAllNotificationSources,
  setNotificationParams,
  resetNotificationApiParams,
  setNewNotificationState
} from './notifications/action'
