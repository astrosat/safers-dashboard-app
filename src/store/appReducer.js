import { combineReducers } from 'redux';
import authReducer from './authentication/reducer';
import userReducer from './user/reducer';
import myProfileReducer from './myprofile/reducer';
import * as actionTypes from './authentication/types';
import commonReducer from './common/reducer';
import dashboardReducer from './dashboard/reducer';
import alertReducer from './alerts/reducer';
import eventAlertReducer from './events/reducer';

import storage from 'redux-persist/lib/storage/session';//or session

const appReducer = combineReducers({
  auth: authReducer,
  user: userReducer,
  myprofile: myProfileReducer,
  common: commonReducer,
  dashboard: dashboardReducer,
  alerts: alertReducer,
  eventAlerts: eventAlertReducer
});

const rootReducer = (state, action) => {
  if (action.type === actionTypes.SIGN_OUT) {
    storage.removeItem('persist:root')
    // storage.removeItem('persist:otherKey')

    return appReducer(undefined, action);
  }
  return appReducer(state, action);
}
export default rootReducer;
