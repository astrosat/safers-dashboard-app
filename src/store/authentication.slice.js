import {
  createAsyncThunk,
  createSlice,
  createSelector,
} from '@reduxjs/toolkit';
import storage from 'redux-persist/lib/storage/session';

import * as api from 'api/base';
import { OK, CREATED } from 'api/constants';
import { endpoints } from 'api/endpoints';
import { setLoading } from 'store/common.slice';
import { setUserInfo } from 'store/user.slice';

import {
  AUTH_BASE_URL,
  AUTH_CLIENT_ID,
  AUTH_TENANT_ID,
  CLIENT_BASE_URL,
  REDIRECT_URL,
} from '../config';
import { setSession, deleteSession, getSession } from '../helpers/authHelper';

const name = 'auth';

const setSessionData = (
  access_token,
  refresh_token,
  user_id,
  rememberMe = false,
  isSSOsession = false,
) => {
  const sessionData = {
    access_token,
    refresh_token,
    userId: user_id,
    isSSOsession,
  };
  setSession(sessionData, rememberMe);
};

export const authenticateOauth2 = createAsyncThunk(
  `${name}/authenticateOauth2`,
  async (authCode, { dispatch, rejectWithValue }) => {
    const response = await api.post(
      endpoints.authentication.oAuth2Authenticate,
      {
        code: authCode,
      },
    );

    dispatch(setLoading({ status: false }));

    if (response.status === OK || response.status === CREATED) {
      const { access_token, expires_in, user_id } = response.data;
      setSessionData(access_token, null, user_id, false, true);

      const userResponse = await api.get(`${endpoints.user.profile}${user_id}`);
      if (userResponse.status === OK) {
        const user = userResponse.data;
        dispatch(setUserInfo(user));
      }
      return {
        user_id,
        expires_in,
      };
    }

    return rejectWithValue({ error: response.data.detail });
  },
);

export const signUpOauth2 = createAsyncThunk(
  `${name}/signUpOauth2`,
  async (userInfo, { rejectWithValue }) => {
    const response = await api.post(
      endpoints.authentication.oAuth2Register,
      userInfo,
    );

    if (response.status === CREATED) {
      window.location = `${CLIENT_BASE_URL}/auth/sign-in`;
      return response.data;
    }

    return rejectWithValue({ error: response.data });
  },
);

export const resetForgottenUserPasswordRequest = createAsyncThunk(
  `${name}/resetForgottenUserPasswordRequest`,
  async (email, { rejectWithValue }) => {
    const response = await api.post(
      endpoints.authentication.forgotPswReset,
      email,
    );

    if (response.status === OK) {
      return response.data;
    }

    return rejectWithValue({ error: response.error });
  },
);

export const resetPassword = createAsyncThunk(
  `${name}/resetPassword`,
  async (data, { rejectWithValue }) => {
    const response = await api.post(endpoints.authentication.resetPsw, {
      ...data,
    });

    if (response.status === OK) {
      return response.data;
    }

    return rejectWithValue({ error: response.data });
  },
);

export const redirectAfterSignOut = () => {
  const params = {
    client_id: AUTH_CLIENT_ID,
    tenant_id: AUTH_TENANT_ID,
    post_logout_redirect_uri: `${CLIENT_BASE_URL}/${REDIRECT_URL}`,
  };
  const urlParams = new URLSearchParams(params).toString();
  window.location = `${AUTH_BASE_URL}/oauth2/logout?${urlParams}`;
};

export const signOut = createAsyncThunk(
  `${name}/signOut`,
  async (_, { rejectWithValue }) => {
    const session = getSession();

    const response = await api.post(endpoints.authentication.oAuth2Logout);

    if (response.status === OK) {
      deleteSession();
      storage.removeItem('persist:root');

      if (session.isSSOsession) {
        redirectAfterSignOut();
      }

      return response.data;
    }

    return rejectWithValue({ error: true });
  },
);

export const refreshOAuthToken = createAsyncThunk(
  `${name}/refreshOAuthToken`,
  async (_, { rejectWithValue }) => {
    const response = await api.post(endpoints.authentication.oAuth2Refresh);

    if (response.status === OK) {
      return response.data;
    }

    return rejectWithValue({ error: response.data });
  },
);

export const initialState = {
  userId: null,
  signUpOauth2Success: false,
  isLoggedIn: false,
  error: false,
  forgotPswresponse: null,
  resetPswRes: null,
  resetPswError: null,
  errorSignIn: false,
  tokenExpiresIn: null,
  tokenLastUpdated: null,
  refreshTokenErr: null,
};

const authenticationSlice = createSlice({
  name,
  initialState,
  reducers: {
    signOutSuccess: state => {
      state = initialState;
    },
  },
  extraReducers: builder => {
    builder
      .addCase(authenticateOauth2.fulfilled, (state, { payload }) => {
        state.userId = payload.user_id;
        state.tokenExpiresIn = payload.expires_in;
        state.isLoggedIn = true;
        state.error = false;
      })
      .addCase(authenticateOauth2.rejected, (state, { payload }) => {
        state.errorSignIn = payload.error;
        state.error = true;
      })
      .addCase(signUpOauth2.fulfilled, state => {
        state.signUpOauth2Success = true;
        state.isLoggedIn = false;
        state.error = false;
      })
      .addCase(signUpOauth2.rejected, (state, { payload }) => {
        state.signUpOauth2Success = false;
        state.error = payload.error;
      })
      .addCase(
        resetForgottenUserPasswordRequest.fulfilled,
        (state, { payload }) => {
          state.forgotPswresponse = payload;
          state.error = false;
        },
      )
      .addCase(
        resetForgottenUserPasswordRequest.rejected,
        (state, { payload }) => {
          state.error = payload.error;
        },
      )
      .addCase(resetPassword.fulfilled, (state, { payload }) => {
        state.resetPswRes = payload;
        state.error = false;
      })
      .addCase(resetPassword.rejected, (state, { payload }) => {
        state.resetPswError = payload.error;
      })
      .addCase(signOut.fulfilled, state => {
        state = initialState;
      })
      .addCase(signOut.rejected, state => {
        state.error = true;
      })
      .addCase(refreshOAuthToken.fulfilled, (state, { payload }) => {
        state.tokenExpiresIn = payload.expires_in;
        state.tokenLastUpdated = Date.now();
        state.error = false;
      })
      .addCase(refreshOAuthToken.rejected, (state, { payload }) => {
        state.refreshTokenErr = payload.error;
      });
  },
});

export const { signOutSuccess } = authenticationSlice.actions;

const baseSelector = state => state?.auth;

export const tokenLastUpdatedSelector = createSelector(
  baseSelector,
  auth => auth?.tokenLastUpdated,
);

export const tokenExpiresInSelector = createSelector(
  baseSelector,
  auth => auth?.tokenExpiresIn,
);

export const isLoggedInSelector = createSelector(
  baseSelector,
  auth => auth?.isLoggedIn,
);

export const errorSelector = createSelector(baseSelector, auth => auth?.error);

export const errorSignInSelector = createSelector(
  baseSelector,
  auth => auth?.errorSignIn,
);

export const resetPasswordResponseSelector = createSelector(
  baseSelector,
  auth => auth?.resetPswRes,
);

export const signUpOauth2SuccessSelector = createSelector(
  baseSelector,
  auth => auth?.signUpOauth2Success,
);

export const forgotPasswordResponseSelector = createSelector(
  baseSelector,
  auth => auth?.forgotPswresponse,
);

export default authenticationSlice.reducer;
