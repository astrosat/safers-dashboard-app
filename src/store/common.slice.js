import {
  createAsyncThunk,
  createSlice,
  createSelector,
} from '@reduxjs/toolkit';

import * as api from 'api/base';
import { OK } from 'api/constants';
import { endpoints } from 'api/endpoints';

import { getDefaultDateRange } from '../utility';

const name = 'common';

export const fetchConfig = createAsyncThunk(
  `${name}/fetchConfig`,
  async (_, { rejectWithValue }) => {
    const response = await api.get(endpoints.common.config);

    if (response.status === OK) {
      return response.data;
    }

    return rejectWithValue({ error: true });
  },
);

export const fetchOrganisations = createAsyncThunk(
  `${name}/fetchOrganisations`,
  async (_, { rejectWithValue }) => {
    const response = await api.get(endpoints.common.organizations);

    if (response.status === OK) {
      return response.data;
    }

    return rejectWithValue({ error: true });
  },
);

export const fetchRoles = createAsyncThunk(
  `${name}/fetchRoles`,
  async (_, { rejectWithValue }) => {
    const response = await api.get(endpoints.common.roles);

    if (response.status === OK) {
      return response.data;
    }

    return rejectWithValue({ error: true });
  },
);

export const fetchTeams = createAsyncThunk(
  `${name}/fetchTeams`,
  async (_, { rejectWithValue }) => {
    const response = await api.get(endpoints.common.teams);

    if (response.status === OK) {
      return response.data;
    }

    return rejectWithValue({ error: true });
  },
);

export const fetchAois = createAsyncThunk(
  `${name}/fetchAois`,
  async (_, { rejectWithValue }) => {
    const response = await api.get(endpoints.aoi.getAll);

    if (response.status === OK) {
      return response.data;
    }

    return rejectWithValue({ error: true });
  },
);

export const fetchTweets = createAsyncThunk(
  `${name}/fetchTweets`,
  async (params, { rejectWithValue }) => {
    const response = await api.get(endpoints.dashboard.getInSitu, params);

    if (response.status === OK) {
      return response.data;
    }

    return rejectWithValue({ error: true });
  },
);

export const initialState = {
  config: undefined,
  orgList: [],
  roleList: [],
  aois: [],
  selectedAoi: null,
  viewState: undefined,
  polygonLayer: undefined,
  isLoading: false,
  loadingMsg: null,
  dateRange: getDefaultDateRange(),
  isDateRangeDisabled: false,
  error: null,
  teamList: [],
};

const commonSlice = createSlice({
  name,
  initialState,
  reducers: {
    setLoading: (state, { payload }) => {
      state.loadingMsg = payload.message;
      state.isLoading = payload.status;
    },
    setSelectedAoi: (state, { payload }) => {
      state.selectedAoi = payload;
    },
    // setViewState: (state, { payload }) => {
    //   state.selectedAoi = payload;
    //   state.isLoading = true;
    // },
    // viewState: (state, { payload }) => {
    //   state.viewState = payload;
    // },
    setPolygonLayer: (state, { payload }) => {
      state.polygonLayer = payload;
    },
    setDateRange: (state, { payload }) => {
      state.dateRange = payload;
    },
    setDateRangeDisabled: (state, { payload }) => {
      state.isDateRangeDisabled = payload;
    },
  },
  extraReducers: builder => {
    builder
      .addCase(fetchConfig.fulfilled, (state, { payload }) => {
        state.config = payload;
        state.error = false;
      })
      .addCase(fetchConfig.rejected, state => {
        state.error = true;
      })
      .addCase(fetchOrganisations.fulfilled, (state, { payload }) => {
        state.orgList = payload;
        state.error = false;
      })
      .addCase(fetchOrganisations.rejected, state => {
        state.error = true;
      })
      .addCase(fetchRoles.fulfilled, (state, { payload }) => {
        state.roleList = payload;
        state.error = false;
      })
      .addCase(fetchRoles.rejected, state => {
        state.error = true;
      })
      .addCase(fetchTeams.fulfilled, (state, { payload }) => {
        state.teamList = payload;
        state.error = false;
      })
      .addCase(fetchTeams.rejected, state => {
        state.error = true;
      })
      .addCase(fetchAois.fulfilled, (state, { payload }) => {
        state.aois = payload;
        state.error = false;
      })
      .addCase(fetchAois.rejected, state => {
        state.error = true;
      });
  },
});

export const {
  setLoading,
  setSelectedAoi,
  setPolygonLayer,
  setDateRange,
  setDateRangeDisabled,
} = commonSlice.actions;

const baseSelector = state => state?.common;

export const isLoadingSelector = createSelector(
  baseSelector,
  common => common?.isLoading,
);

export const loadingMessageSelector = createSelector(
  baseSelector,
  common => common?.loadingMsg,
);

export const configSelector = createSelector(
  baseSelector,
  common => common?.config,
);

export const isDateRangeDisabledSelector = createSelector(
  baseSelector,
  common => common?.isDateRangeDisabled,
);

export const dateRangeSelector = createSelector(
  baseSelector,
  common => common?.dateRange,
);

export const aoisSelector = createSelector(
  baseSelector,
  common => common?.aois,
);

export const selectedAoiSelector = createSelector(
  baseSelector,
  common => common?.selectedAoi,
);

export const organisationsSelector = createSelector(
  baseSelector,
  common => common?.orgList,
);

export const rolesSelector = createSelector(
  baseSelector,
  common => common?.roleList,
);

export const teamsSelector = createSelector(
  baseSelector,
  common => common?.teamList,
);

export default commonSlice.reducer;
