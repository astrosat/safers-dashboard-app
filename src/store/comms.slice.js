import {
  createAsyncThunk,
  createSlice,
  createSelector,
} from '@reduxjs/toolkit';
import toastr from 'toastr';

import * as api from 'api/base';
import { OK } from 'api/constants';
import { endpoints } from 'api/endpoints';
import { getFilteredRecords } from 'pages/Chatbot/filter';

const name = 'comms';

export const fetchComms = createAsyncThunk(
  `${name}/fetchComms`,
  async (
    { options, feFilters = null, isPolling = false },
    { rejectWithValue },
  ) => {
    const response = await api.get(endpoints.chatbot.comms.getAll, options);

    if (response.status === OK) {
      return {
        isPolling,
        feFilters,
        data: response.data,
      };
    }

    return rejectWithValue({ error: true });
  },
);

export const createComms = createAsyncThunk(
  `${name}/createComms`,
  async (comms, { rejectWithValue }) => {
    const response = await api.post(endpoints.chatbot.comms.createMsg, comms);

    if (response.status === OK) {
      toastr.success(response.data.msg, '');
      return response.data;
    } else {
      return rejectWithValue({ error: true });
    }
  },
);

export const initialState = {
  allComms: [],
  pollingData: [],
  sortByDate: 'desc',
  alertSource: 'all',
  error: false,
  filteredComms: null,
};

const commsSlice = createSlice({
  name,
  initialState,
  reducers: {
    resetCommsResponseState: state => {
      state.error = false;
    },
    refreshData: (state, { payload }) => {
      state.allComms = payload;
      state.filteredComms = null;
      state.error = false;
    },
    setFilteredComms: (state, { payload }) => {
      state.filteredComms = payload;
      state.error = false;
    },
  },
  extraReducers: builder => {
    builder
      .addCase(fetchComms.fulfilled, (state, { payload }) => {
        if (payload.isPolling) {
          state.pollingData = payload.data;
          state.error = false;
        } else {
          const { target, status, sortOrder } = payload.feFilters ?? {};
          const filters = { target, status };
          const sort = { fieldName: 'start', order: sortOrder };
          const filteredComms = getFilteredRecords(payload.data, filters, sort);

          state.filteredComms = filteredComms;
          state.allComms = payload.data;
          state.error = false;
        }
      })
      .addCase(fetchComms.rejected, state => {
        state.error = true;
      })
      .addCase(createComms.fulfilled, state => {
        state.error = false;
      })
      .addCase(createComms.rejected, state => {
        state.error = true;
      });
  },
});

export const { resetCommsResponseState, refreshData, setFilteredComms } =
  commsSlice.actions;

const baseSelector = state => state?.comms;

export const allCommsSelector = createSelector(
  baseSelector,
  comms => comms?.allComms,
);

export const filteredCommsSelector = createSelector(
  baseSelector,
  comms => comms?.filteredComms,
);

export const commsPollingDataSelector = createSelector(
  baseSelector,
  comms => comms?.pollingData,
);

export default commsSlice.reducer;
