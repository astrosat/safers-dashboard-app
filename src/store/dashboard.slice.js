import {
  createAsyncThunk,
  createSlice,
  createSelector,
} from '@reduxjs/toolkit';

import * as api from 'api/base';
import { OK } from 'api/constants';
import { endpoints } from 'api/endpoints';

const name = 'dashboard';

export const fetchStats = createAsyncThunk(
  `${name}/fetchStats`,
  async (params, { rejectWithValue }) => {
    const response = await api.get(endpoints.dashboard.getStats, params);

    if (response.status === OK) {
      return response.data;
    }

    return rejectWithValue({ error: true });
  },
);

export const fetchWeatherStats = createAsyncThunk(
  `${name}/fetchWeatherStats`,
  async (params, { rejectWithValue }) => {
    const response = await api.get(endpoints.dashboard.getWeatherStats, params);

    if (response.status === OK) {
      return response.data;
    }

    return rejectWithValue({ error: true });
  },
);

export const fetchWeatherVariables = createAsyncThunk(
  `${name}/fetchWeatherVariables`,
  async (params, { rejectWithValue }) => {
    const response = await api.get(
      endpoints.dashboard.getWeatherVariables,
      params,
    );

    if (response.status === OK) {
      return response.data;
    }

    return rejectWithValue({ error: true });
  },
);

export const fetchInSituMedia = createAsyncThunk(
  `${name}/fetchInSituMedia`,
  async (params, { rejectWithValue }) => {
    const response = await api.get(endpoints.dashboard.getInSitu, params);

    if (response.status === OK) {
      return response.data;
    }

    return rejectWithValue({ error: true });
  },
);

export const fetchTweets = createAsyncThunk(
  `${name}/fetchTweets`,
  async (params, { rejectWithValue }) => {
    const response = await api.get(endpoints.dashboard.getInSitu, params);

    if (response.status === OK) {
      return response.data;
    }

    return rejectWithValue({ error: true });
  },
);

export const initialState = {
  stats: {},
  weatherStats: {},
  weatherVariables: [],
  inSituMedia: [],
  tweets: [],
  error: null,
};

const dashboardSlice = createSlice({
  name,
  initialState,
  reducers: {},
  extraReducers: builder => {
    builder
      .addCase(fetchStats.fulfilled, (state, { payload }) => {
        state.stats = payload;
        state.error = false;
      })
      .addCase(fetchStats.rejected, state => {
        state.error = true;
      })
      .addCase(fetchWeatherStats.fulfilled, (state, { payload }) => {
        state.weatherStats = payload;
        state.error = false;
      })
      .addCase(fetchWeatherStats.rejected, state => {
        state.error = true;
      })
      .addCase(fetchWeatherVariables.fulfilled, (state, { payload }) => {
        state.weatherVariables = payload;
        state.error = false;
      })
      .addCase(fetchWeatherVariables.rejected, state => {
        state.error = true;
      })
      .addCase(fetchInSituMedia.fulfilled, (state, { payload }) => {
        state.inSituMedia = payload;
        state.error = false;
      })
      .addCase(fetchInSituMedia.rejected, state => {
        state.error = true;
      })
      .addCase(fetchTweets.fulfilled, (state, { payload }) => {
        state.tweets = payload;
        state.error = false;
      })
      .addCase(fetchTweets.rejected, state => {
        state.error = true;
      });
  },
});

const baseSelector = state => state?.dashboard;

export const tweetsSelector = createSelector(
  baseSelector,
  dashboard => dashboard?.tweets,
);

export const statsSelector = createSelector(
  baseSelector,
  dashboard => dashboard?.stats,
);

export const weatherStatsSelector = createSelector(
  baseSelector,
  dashboard => dashboard?.weatherStats,
);

export default dashboardSlice.reducer;
