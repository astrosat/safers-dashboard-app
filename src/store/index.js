import { createStore, compose, applyMiddleware } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage/session';//or session
import thunk from 'redux-thunk';

import reducers from './appReducer';
// eslint-disable-next-line no-unused-vars
import logger from './middleware/logger';

const persistConfig = {
  key: 'root',
  storage,
  whitelist: ['auth', 'user', 'common'],
};

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const persistedReducer = persistReducer(persistConfig, reducers);

const store = createStore(
  persistedReducer,
  composeEnhancers(applyMiddleware(
    thunk,
    // logger
  ))
);

export const persistor = persistStore(store);

export default store;
