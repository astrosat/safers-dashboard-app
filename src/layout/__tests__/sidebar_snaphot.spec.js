import React from 'react';

import { cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import ShallowRenderer from 'react-test-renderer/shallow';

import store from '../../store';
import {
  mockedDispatch,
  mockedHref,
  mockedNavigator,
  mockedSelector,
} from '../../TestUtils';
import Sidebar from '../Sidebar';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // technically it passes without this too, but I'm not sure if its there for other tests to use the real thing so I left it in
  useNavigate: () => mockedNavigator,
  useHref: () => mockedHref,
}));
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'), // technically it passes without this too, but I'm not sure if its there for other tests to use the real thing so I left it in
  useSelector: () => mockedSelector,
  useDispatch: () => mockedDispatch,
}));

afterEach(cleanup);

describe('SideBar Component', () => {
  describe('Renders', () => {
    it('Should match snapshot', () => {
      const renderer = new ShallowRenderer();
      const result = renderer.render(
        <Provider store={store}>
          <Sidebar />
        </Provider>,
      );
      expect(result).toMatchSnapshot();
    });
  });
});
