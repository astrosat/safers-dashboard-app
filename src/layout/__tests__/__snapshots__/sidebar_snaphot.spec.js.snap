// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar Component Renders Should match snapshot 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Object {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "onStateChange": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Memo(Connect(Sidebar)) />
</Context.Provider>
`;
